apps/web/src/components/common/TokenIcon/index.tsx

```diff
@@ -15,15 +18,26 @@ const TokenIcon = ({
   size?: number
   fallbackSrc?: string
 }): ReactElement => {
+  let crossOrigin = false
+  let src = logoUri
+
+  if (logoUri?.startsWith(COINGECKO_URL)) {
+    src = logoUri?.replace(COINGECKO_THUMB, COINGECKO_SMALL)
+    crossOrigin = true
+  }
+
   return (
     <ImageFallback
-      src={logoUri}
+      src={src}
       alt={tokenSymbol}
       fallbackSrc={fallbackSrc || FALLBACK_ICON}
       height={size}
       className={css.image}
+      loading="lazy"
+      referrerPolicy={crossOrigin ? 'no-referrer' : undefined}
+      crossOrigin={crossOrigin ? 'anonymous' : undefined}
     />
   )
 }
 
-export default TokenIcon
+export default memo(TokenIcon)
```

Comment:
small question, do we really need to use memo here? Because since this component just renders an image, I'm don't think it would be a problem in this case

Reply:
Because I added a replace there. There can be hundreds of icons on each history page so it’s better not to rerender them.